tf_quant_finance/black_scholes/asian_prices_test.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for asian_prices."""
+
+import functools
+
+from absl.testing import parameterized
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tensorflow_probability as tfp
+
+import tf_quant_finance as tff
+from tensorflow.python.framework import test_util  # pylint: disable=g-direct-tensorflow-import
+
+
+@test_util.run_all_in_graph_and_eager_modes
+class AsianPriceTest(parameterized.TestCase, tf.test.TestCase):
+  """Tests for methods for the asian pricing module"""
+
+  def test_option_prices(self):
+    """Tests for methods for the asian pricing module. Results come from
+    'Implementing Derivatives Model' by Clewlow, Strickland p.118-123, via the
+    QuantLib test suite
+    """
+    spots = np.array([100.0])
+    dividend_rates = np.array([0.03])
+    discount_rates = np.array([0.06])
+    volatilities = np.array([0.2])
+    strikes = np.array([100.0])
```

Comment:
let's also make it work with scalar inputs:  spots = 100, divident_rates = 0.03, ...
you might need to check rank of inputs (e.g., spots.shape.rank) and see if that is zero, add an extra dimension (e.g., spots = spots[tf.newaxis])

Reply:
Added a test to do this